src/mlpack/methods/ann/make_alias.hpp

```diff
@@ -15,42 +16,150 @@
 #define MLPACK_METHODS_ANN_MAKE_ALIAS_HPP
 
 #include <mlpack/prereqs.hpp>
+#include <bandicoot>
```

Comment:
I think that we shouldn't actually include Bandicoot anywhere in the mlpack headers since it's not a strict dependency. I had imaged a user would use bandicoot+mlpack in their source like this:

```c++
#include <bandicoot>
#include <mlpack.hpp> // detects that bandicoot is available and sets MLPACK_HAS_COOT or similar
```

I'm open to other possibilities if you have other ideas.  But basically this would just mean there needed to be a check somewhere in prereqs.hpp or similar that would set `MLPACK_HAS_COOT` accordingly, and then we would need to wrap any usage of anything Bandicoot-specific (e.g. any explicit naming of `coot::`) in a macro guard.

Getting the tests to automatically test against Bandicoot will be a little trickier and involve some CMake changes, but we don't have to do that in this PR for sure. :+1:

Reply:
For sure, I agree, less dependencies is better, you know me, I am not too fan of dependencies :+1: 
Actually, by doing this I was thinking, that once we have made mlpack entirely generic, there is no reason to include armadillo internally at all, we can remove it, and push it to the user side how to include it.

Reply:
I probably add it there by error while testing,

Reply:
Actually, this is very hard to do, so maybe in the very far future, it also worth considering

Reply:
We do need a linear algebra library; I don't think we can remove Armadillo as a dependency.  Especially since Bandicoot itself depends on Armadillo.